### PR TITLE
[HttpKernel] Renamed the argument resolver tag

### PR DESCRIPTION
**Pull request by linaori**
| Q | A |
| --- | --- |
| Branch? | master |
| Bug fix? | no |
| New feature? | no |
| BC breaks? | not if merged before 3.1 |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | ~ |
| License | MIT |
| Doc PR | ~ |

Changed as discussed several times: https://github.com/symfony/symfony/pull/18510#issuecomment-208778875, https://github.com/symfony/symfony-docs/pull/6422#issuecomment-208453616.

### CODE REVIEW

**Comment by stof** [comment]
if this does not require any change in the testsuite, it looks like we are missing some tests...

**Comment by iltar** [comment]
@stof this is functionally tested only (implicitly), I can add a compiler pass unit-test though.

**Comment by javiereguiluz** [comment]
👍

**Comment by iltar** [comment]
Added a test and named the compiler pass properly, it was still having "Controller" which is not part of the tag anymore.

**Comment by nicolas-grekas** [comment]
:+1:

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/ArgumentValueResolverPass.php [hunk 1]
```diff
@@ -20,7 +20,7 @@
  *
  * @author Iltar van der Berg <kjarli@gmail.com>
  */
-class ControllerArgumentValueResolverPass implements CompilerPassInterface
+class ArgumentValueResolverPass implements CompilerPassInterface
```
Comment (fabpot) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/ArgumentValueResolverPass.php]: This rename looks wrong to me. Compiler passes is in the `Symfony\Bundle\FrameworkBundle\DependencyInjection\Compiler` namespace, which gives no hints as to what Argument refers to.

**Comment by fabpot** [comment]
👎 I'm all for having shorter names, but they need to be self-descriptive as well. Here, we're not talking about any argument, but only controller arguments.

**Comment by javiereguiluz** [comment]
Other alternatives considered:

* `controller.value_resolver`
* `controller.argument_value_resolver`
* `controller.argument_value`
* `controller.argument_resolver`

**Comment by xabbuh** [comment]
If we were to change this, I would vote for `controller.argument_value_resolver`.

**Comment by iltar** [comment]
@xabbuh That's actually the same length as before:
 - `controller_argument.value_resolver`
 - `controller.argument_value_resolver`

**Comment by xabbuh** [comment]
@iltar That's true. But it still conveys the proper meaning while making the namespace more clear imo. To me all other suggestions lose too much information.

**Comment by iltar** [comment]
That's true, in that case I will also name the class the same as before again.

**Comment by iltar** [comment]
Rebased to a clean commit, if everyone agrees on the name, I will update my other PRs

**Comment by fabpot** [comment]
:+1:

**Comment by xabbuh** [comment]
Why do you mock the DI related classes in the test? Imo you should rather use the real classes and check that the definitions are modified as expected.

**Comment by iltar** [comment]
@xabbuh I can do that, this was simply copied from the other tests

File: src/Symfony/Bundle/FrameworkBundle/Tests/DependencyInjection/Compiler/ControllerArgumentValueResolverPassTest.php [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        );
+
+        $expected = array(
+            new Reference('n1'),
+            new Reference('n2'),
+            new Reference('n3'),
+        );
+
+        $definition = new Definition(ArgumentResolver::class, array(null, array()));
+        $container = $this->getMock(ContainerBuilder::class, array('findTaggedServiceIds'));
```
Comment (xabbuh) [inline src/Symfony/Bundle/FrameworkBundle/Tests/DependencyInjection/Compiler/ControllerArgumentValueResolverPassTest.php]: You can just create a `ContainerBuilder` instance here (like, for example, done in the `FormPassTest` and `CachePoolPassTest`).

Comment (xabbuh) [inline src/Symfony/Bundle/FrameworkBundle/Tests/DependencyInjection/Compiler/ControllerArgumentValueResolverPassTest.php]: and the same for the other test methods as well

**Comment by iltar** [comment]
@xabbuh, no more mocking ;)

**Comment by fabpot** [comment]
Thank you @iltar.

**Pull request merged**